numpyro/util.py

```diff
@@ -172,9 +172,16 @@ def progress_bar_factory(num_samples):
 
     tqdm_bars = {}
 
+    if num_samples > 20:
```

Comment:
This (along with the change below) fixes 2 bugs:
1. if `num_samples` < 20 you now don't get a modulo zero error.
2. The progress bar now works if 20 doesn't divide the `num_samples` (before it overshot the total number of iterations)

Reply:
thanks, the logic looks great!